Tests/PackageRegistryTests/RegistryConfigurationTests.swift

```diff
@@ -186,4 +377,171 @@ final class RegistryConfigurationTests: XCTestCase {
+                            "trustedRootCertificatesPath": "/foo/roots",
+                            "validationChecks": {
+                                "certificateRevocation": "allowSoftFail"
+                            }
+                        }
+                    },
+                    "other.example.com": {
+                        "signing": {
+                            "onUntrustedCertificate": "error"
+                        }
+                    }
+                }
+            },
+            "version": 1
+        }
+        """#
+
+        let configuration = try decoder.decode(RegistryConfiguration.self, from: json)
+
+        let package = PackageIdentity.plain("mona.LinkedList")
+        let registry = Registry(url: URL(string: "https://packages.example.com")!)
+        let signing = try configuration.signing(for: package, registry: registry)
+
+        XCTAssertEqual(signing.onUnsigned, .prompt)
+        XCTAssertEqual(signing.onUntrustedCertificate, .warn)
+        XCTAssertEqual(signing.trustedRootCertificatesPath, "/foo/roots")
+        XCTAssertEqual(signing.includeDefaultTrustedRootCertificates, true)
+        XCTAssertEqual(signing.validationChecks?.certificateExpiration, .disabled)
+        XCTAssertEqual(signing.validationChecks?.certificateRevocation, .allowSoftFail)
+    }
+
+    func testGetSigning_scopeOverride() throws {
+        let json = #"""
+        {
+            "registries": {},
+            "security": {
+                "scopeOverrides": {
+                    "mona": {
+                        "signing": {
+                            "trustedRootCertificatesPath": "/mona/roots"
+                        }
+                    },
+                    "foo": {
+                        "signing": {
+                            "trustedRootCertificatesPath": "/foo/roots"
+                        }
+                    }
+                }
+            },
+            "version": 1
+        }
+        """#
+
+        let configuration = try decoder.decode(RegistryConfiguration.self, from: json)
+
+        let package = PackageIdentity.plain("mona.LinkedList")
+        let registry = Registry(url: URL(string: "https://packages.example.com")!)
+        let signing = try configuration.signing(for: package, registry: registry)
+
+        XCTAssertEqual(signing.onUnsigned, .prompt)
+        XCTAssertEqual(signing.onUntrustedCertificate, .prompt)
+        XCTAssertEqual(signing.trustedRootCertificatesPath, "/mona/roots")
+        XCTAssertEqual(signing.includeDefaultTrustedRootCertificates, true)
+        XCTAssertEqual(signing.validationChecks?.certificateExpiration, .disabled)
+        XCTAssertEqual(signing.validationChecks?.certificateRevocation, .disabled)
+    }
+
+    func testGetSigning_packageOverride() throws {
+        let json = #"""
+        {
+            "registries": {},
+            "security": {
+                "packageOverrides": {
```

Comment:
do we need a test that checks the various levels of overrides / merge behavior?

Reply:
This is what this test and others (e.g., `testGetSigning_scopeOverride`) do.

Reply:
right. I was thinking a test that checks package override of scope, scope override of registry, and registry override of global all in the same config - ie stacked up on top of each other. maybe already covered and I missed it

Reply:
amended https://github.com/apple/swift-package-manager/pull/6159/commits/5d4b3f00ff2dedb99a19ac9e0699dfaf49b2113c